src/DagGenerator.cpp

```diff
@@ -320,6 +320,9 @@ static bool WriteNodes(
             case ActionType::kWriteTextFile:            
                 WriteStringPtr(node_data_seg, writetextfile_payloads_seg, writetextfile_payload);
                 break;
+            case ActionType::kCopyFile:
+                BinarySegmentWriteNullPointer(node_data_seg);
```

Comment:
maybe write a "internal copy" string here, to not break consumers of the structured log that can up to this point have been relying on the action always existing for a noderesult message

Reply:
We will still produce `action: null` into the structured log, and we will still produce a valid annotation. As far as I know the only consumers of the structured log are Bee (which I can test and fix in the Unity PR) and IncrementalBuildPipeline (which I would have expected to be relying on the annotation, but not the action).

Reply:
you're right